src/Extension.Tests/ChallengeTests.cs

```diff
@@ -112,10 +98,7 @@ public async Task challenge_tracks_evaluations_in_submission_history()
         {
             var capturedEvaluation = new List<ChallengeEvaluation>();
             var lesson = new Lesson();
-            using var kernel = new CompositeKernel
-            {
-                new CSharpKernel()
-            }.UseLessonEvaluateMiddleware(lesson);
+            using var kernel = CreateKernel(lesson);
             var challenge = GetEmptyChallenge();
             await lesson.StartChallengeAsync(challenge);
```

Comment:
same here, starting a challenge that is not part of a lesson should not be allowed